File: SocialChallenge/Model/Persistent/ManagedObjects/JSCPostPage.m
//
//  JSCPostPage.m
//  SocialChallenge
//
//  Created by Javier Laguna on 31/01/2016.
//
//

#import "JSCPostPage.h"

#import "NSManagedObjectContext+CDSRetrieval.h"

@implementation JSCPostPage

#pragma mark - LastPage

+ (JSCPostPage *)fetchLastPageInContext:(NSManagedObjectContext *)managedObjectContext
{
    NSSortDescriptor *retrievedPagesSortDescriptor = [NSSortDescriptor sortDescriptorWithKey:@"index"
                                                                                   ascending:NO];
    
    NSArray *sortDescriptor = @[retrievedPagesSortDescriptor];
    
    return (JSCPostPage *) [managedObjectContext cds_retrieveFirstEntryForEntityClass:[JSCPostPage class]
                                                                            predicate:nil
                                                                      sortDescriptors:sortDescriptor];
}

@end

File: SocialChallenge/Model/Persistent/ManagedObjects/JSCPostPage.swift
//
//  JSCPostPage.swift
//  
//
//  Created by Javier Laguna on 04/03/2016.
//
//

import Foundation
import CoreData
import CoreDataServices

@objc(JSCPostPage)

class JSCPostPage: NSManagedObject {
    
    /**
    Retrieves the last page of posts stored in the context.
    
    @param context in which we want to perform the search.
    
    @return JSCPostPage - instace of the JSCPostPage class
    */
    class func fetchLastPageInContext(managedObjectContext : NSManagedObjectContext) -> JSCPostPage {
        
        let retrievedPagesSortDescriptor = NSSortDescriptor.init(key : "index", ascending : false)
        
        let sortDescriptors = [retrievedPagesSortDescriptor]
        
        return managedObjectContext.cds_retrieveFirstEntryForEntityClass(JSCPostPage.self, predicate : nil, sortDescriptors : sortDescriptors) as! JSCPostPage
    }
    
}
